autoscale/src/main/java/com/sequenceiq/periscope/service/ScalingTriggerService.java

```diff
@@ -0,0 +1,101 @@
+    private ClusterService clusterService;
+
+    @Inject
+    private ScalingTriggerRepository scalingTriggerRepository;
+
+    @Inject
+    private RegionAwareCrnGenerator regionAwareCrnGenerator;
+
+    public ScalingTrigger create(Cluster cluster, TriggerStatus triggerStatus, String yarnMetricsResponse, long creationTimestamp) {
+        ScalingTrigger scalingTrigger = new ScalingTrigger();
+        scalingTrigger.setCluster(cluster);
+        scalingTrigger.setTriggerCrn(createTriggerCrn(Crn.safeFromString(cluster.getClusterPertain().getUserCrn()).getAccountId()));
+        scalingTrigger.setTriggerStatus(triggerStatus);
+        scalingTrigger.setYarnMetrics(yarnMetricsResponse);
+        scalingTrigger.setStartTime(new Date(creationTimestamp));
+        return scalingTriggerRepository.save(scalingTrigger);
+    }
+
+    public void updateWithFlowIdAndTriggerStatus(FlowIdentifier flowIdentifier, String triggerCrn, TriggerStatus triggerStatus) {
+        ScalingTrigger scalingTrigger = findByCrn(triggerCrn);
+        scalingTrigger.setFlowId(flowIdentifier.getPollableId());
+        scalingTrigger.setTriggerStatus(triggerStatus);
+        save(scalingTrigger);
+    }
+
+    public ScalingTrigger findByCrn(String triggerCrn) {
+        return scalingTriggerRepository.findByTriggerCrn(triggerCrn).orElseThrow(notFound("ScalingTrigger", triggerCrn));
+    }
+
+    public List<Long> findAllIdsOlderThanDurationForCluster(Cluster cluster, long durationInMinutes) {
+        Date startTimeBefore = new Date(Instant.now().minus(durationInMinutes, ChronoUnit.MINUTES).toEpochMilli());
+        return scalingTriggerRepository.findAllByClusterWithStartTimeBefore(cluster.getId(), startTimeBefore)
+                .stream().map(ScalingTrigger::getId).collect(Collectors.toList());
+    }
+
+    public List<ScalingTrigger> findAllScalingTriggersWithStatusesForCluster(Cluster cluster, Collection<TriggerStatus> statuses) {
+        return scalingTriggerRepository.findAllByClusterAndInTriggerStatuses(cluster.getId(), statuses);
+    }
+
+    public List<Long> findAllIdsOfSuccessfulScalingTriggersForCluster(Cluster cluster) {
+        return findAllScalingTriggersWithStatusesForCluster(cluster, newArrayList(UPSCALE_TRIGGER_SUCCESS, DOWNSCALE_TRIGGER_SUCCESS))
+                .stream().map(ScalingTrigger::getId).collect(Collectors.toList());
+    }
+
+    public List<ScalingTrigger> findAllByTriggerStatusBetweenIntervalForCluster(Cluster cluster, TriggerStatus status, long timestampFrom,
+            long timestampUntil) {
+        Date startTimeFrom = new Date(timestampFrom);
+        Date startTimeUntil = new Date(timestampUntil);
+        return scalingTriggerRepository.findAllByClusterAndTriggerStatusBetweenInterval(cluster.getId(), status, startTimeFrom, startTimeUntil);
+    }
+
+    public void deleteScalingTriggers(Cluster cluster, Set<Long> triggerIds) {
+        cluster.setScalingTriggers(detachScalingTriggersFromCluster(cluster, triggerIds));
```

Comment:
The invocation of detachScalingTriggersFromCluster - can this be skipped completely, if we skip any changes to Cluster itself, and allow the Trigger tracking to be completely independent?